kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServiceProperties.java

```diff
@@ -145,6 +145,97 @@ public void setIp4DnsServers(String ifname, String dnsServers) {
+        return Optional.ofNullable(
+                (Integer) this.properties.get(String.format(NET_INTERFACE_CONFIG_IP6_WAN_PRIORITY, ifname)));
+    }
+
+    public void setIp6WanPriority(String ifname, int priority) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_WAN_PRIORITY, ifname), priority);
+    }
+
+    public Optional<String> getIp6AddressMethod(String ifname) {
+        return getNonEmptyStringProperty(
+                this.properties.get(String.format(NET_INTERFACE_CONFIG_IP6_ADDRESS_METHOD, ifname)));
+    }
+
+    public void setIp6AddressMethod(String ifname, String addressMethod) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_ADDRESS_METHOD, ifname), addressMethod);
+    }
+
+    public Optional<String> getIp6AddressGenMode(String ifname) {
+        return getNonEmptyStringProperty(
+                this.properties.get(String.format(NET_INTERFACE_CONFIG_IP6_ADDR_GEN_MODE, ifname)));
+    }
+
+    public void setIp6AddressGenMode(String ifname, String addrGenMode) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_ADDR_GEN_MODE, ifname), addrGenMode);
+    }
+
+    public Optional<String> getIp6Privacy(String ifname) {
+        return getNonEmptyStringProperty(
+                this.properties.get(String.format(NET_INTERFACE_CONFIG_IP6_PRIVACY, ifname)));
+    }
+
+    public void setIp6Privacy(String ifname, String privacy) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_PRIVACY, ifname), privacy);
+    }
+
+    public String getIp6Address(String ifname) {
+        return (String) this.properties.getOrDefault(String.format(NET_INTERFACE_CONFIG_IP6_ADDRESS, ifname), "");
+    }
+
+    public void setIp6Address(String ifname, String address) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_ADDRESS, ifname), address);
+    }
+
+    public Optional<Integer> getIp6Netmask(String ifname) {
+        return Optional
+                .ofNullable((Integer) this.properties.get(String.format(NET_INTERFACE_CONFIG_IP6_NETMASK, ifname)));
+    }
+
+    public void setIp6Netmask(String ifname, int netmask) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_NETMASK, ifname), netmask);
+    }
+
+    public String getIp6Gateway(String ifname) {
+        return (String) this.properties.getOrDefault(String.format(NET_INTERFACE_CONFIG_IP6_GATEWAY, ifname), "");
+    }
+
+    public void setIp6Gateway(String ifname, String gateway) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_GATEWAY, ifname), gateway);
+    }
+
+    public String getIp6DnsServers(String ifname) {
+        return (String) this.properties.getOrDefault(String.format(NET_INTERFACE_CONFIG_IP6_DNS_SERVERS, ifname), "");
+    }
+
+    public void setIp6DnsServers(String ifname, String dnsServers) {
+        this.properties.put(String.format(NET_INTERFACE_CONFIG_IP6_DNS_SERVERS, ifname), dnsServers);
+    }
```

Comment:
Actually, we represent the DNS servers as a list of strings. See

https://github.com/eclipse/kura/blob/8566aa9cd5feaf9165d6e6d2e352a1b0f491f6e0/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java#L242-L243

Reply:
Actually the `props.getOptStringList` does the conversion into `List<String>`, which is especting a comma-separated string.

Reply:
Yeah I know. I deduce that, in this code, you don't care about what the comma-separated-string is representing. You just want the comma-separated-string. Is this correct?

Reply:
Correct